src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,46 +194,66 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            var childrenAsConcurrentQueue = Children as ConcurrentQueue<ITestResult>;
+            if (childrenAsConcurrentQueue != null)
+                childrenAsConcurrentQueue.Enqueue(result);
+            else
+#endif
+            {
+                var childrenAsIList = Children as IList<ITestResult>;
+                if (childrenAsIList != null)
+                    childrenAsIList.Add(result);
+                else
+                    throw new NotSupportedException("cannot add results to Children");
```

Comment:
Bear in mind we are talking about an internal implementation class, albeit only enforced by convention. I don't envision there being further derivations, since the distinction is between a single result and a composite result.

As a matter of philosophy, we have not used `sealed` in the past even on classes not designed for inheritance. Some people advocate doing that. Our philosophy has been "If you mess with our internals, you take your chances." If you and @rprouse perfer to seal classes that present risks if incorrectly extended, I can go along.